pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -51,8 +51,11 @@ type SpdyRoundTripper struct {
 	// conn is the underlying network connection to the remote server.
 	conn net.Conn
 
-	// Dialer is the dialer used to connect.  Used if non-nil.
-	Dialer *net.Dialer
+	// dialer handle dialing for this round trip
+	dialer Dialer
```

Comment:
I assume this was exported for other tests... should probably keep it that way

I don't disagree that this could use a refactor, but I'd like to minimize non-test changes until after 1.2 gets cut. To do that, I'd just change this from a `*net.Dialer` to a `func(network, address string) (net.Conn, error)`, and change any callers to set it to the `.Dial` func of the net.Dialer they were previously passing